Automattic/pocket-casts-ios
# Episodes Autoplay: extract SQL queries to its own class

This PR extracts the SQL queries that power the following screens:

* Podcast
* Filters
* Downloads
* Files
* Starred
* Listening History

This step is necessary for the autoplay, given we need a way to retrieve the same list of episodes so we can accurately play the next one.

No new features are added to this PR.

## To test

Ps.: you can use `trunk` or the main app running in another simulator/device so you can compare the list of episodes.

1. Run the app (I recommend running into on a logged account, with Filters, Downloaded files, Starred, etc)
2. Go to Discover and check any random podcast
3. ✅ The episodes should appear correctly
4. Do the same in your "Podcasts" tab
5. ✅ Check that episodes appear correctly
6. Go to "Profile" and tap "Downloads"
7. ✅ Check that episodes appear correctly
8. Go back to "Profile" and tap "Files"
9. ✅ Check that your files appear correctly
10. Go back to "Profile" and tap "Starred"
11. ✅ Check that your episodes appear correctly
12. Go back to "Profile" and tap "Listening History"
11. ✅ Check that your played episodes appear correctly

## Checklist
